src/server/hset_family.cc

```diff
@@ -1060,27 +1061,52 @@ using CI = CommandId;
 
 #define HFUNC(x) SetHandler(&HSetFamily::x)
 
+namespace {
+namespace Acl {
+namespace Cat = AclCategory;
+constexpr uint32_t kHDel = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHLen = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHExists = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHGet = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHGetAll = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHMGet = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHMSet = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHIncrBy = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHIncrByFloat = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHKeys = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHRandField = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHScan = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHSet = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHSetEx = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHSetNx = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHStrLen = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHVals = Cat::READ | Cat::HASH | Cat::SLOW;
+}  // namespace Acl
+}  // namespace
+
 void HSetFamily::Register(CommandRegistry* registry) {
-  *registry << CI{"HDEL", CO::FAST | CO::WRITE, -3, 1, 1, 1}.HFUNC(HDel)
-            << CI{"HLEN", CO::FAST | CO::READONLY, 2, 1, 1, 1}.HFUNC(HLen)
-            << CI{"HEXISTS", CO::FAST | CO::READONLY, 3, 1, 1, 1}.HFUNC(HExists)
-            << CI{"HGET", CO::FAST | CO::READONLY, 3, 1, 1, 1}.HFUNC(HGet)
-            << CI{"HGETALL", CO::FAST | CO::READONLY, 2, 1, 1, 1}.HFUNC(HGetAll)
-            << CI{"HMGET", CO::FAST | CO::READONLY, -3, 1, 1, 1}.HFUNC(HMGet)
-            << CI{"HMSET", CO::WRITE | CO::FAST | CO::DENYOOM, -4, 1, 1, 1}.HFUNC(HSet)
-            << CI{"HINCRBY", CO::WRITE | CO::DENYOOM | CO::FAST, 4, 1, 1, 1}.HFUNC(HIncrBy)
-            << CI{"HINCRBYFLOAT", CO::WRITE | CO::DENYOOM | CO::FAST, 4, 1, 1, 1}.HFUNC(
-                   HIncrByFloat)
-            << CI{"HKEYS", CO::READONLY, 2, 1, 1, 1}.HFUNC(HKeys)
-
-            // TODO: add options support
-            << CI{"HRANDFIELD", CO::READONLY, 2, 1, 1, 1}.HFUNC(HRandField)
-            << CI{"HSCAN", CO::READONLY, -3, 1, 1, 1}.HFUNC(HScan)
-            << CI{"HSET", CO::WRITE | CO::FAST | CO::DENYOOM, -4, 1, 1, 1}.HFUNC(HSet)
-            << CI{"HSETEX", CO::WRITE | CO::FAST | CO::DENYOOM, -5, 1, 1, 1}.SetHandler(HSetEx)
-            << CI{"HSETNX", CO::WRITE | CO::DENYOOM | CO::FAST, 4, 1, 1, 1}.HFUNC(HSetNx)
-            << CI{"HSTRLEN", CO::READONLY | CO::FAST, 3, 1, 1, 1}.HFUNC(HStrLen)
-            << CI{"HVALS", CO::READONLY, 2, 1, 1, 1}.HFUNC(HVals);
+  *registry
```

Comment:
another idea - to introduce manipulators:
```
 *registry << EnableCat(HASH) << ....
<< DisableCat(HASH)
```

Reply:
That's a nice one and I was thinking of it for acl commands/subcommands but for this case it will add a lot of verbosity. At least with the `constexpr` variables we have everything in one place and we just pass a small (meaning in letters small) argument to the constructor.